File: android/src/main/java/com/tpstreams/LicenseDurationUtils.kt
package com.tpstreams

object LicenseDurationUtils {
    const val DEFAULT_LICENSE_EXPIRY_SECONDS = 15L * 24 * 60 * 60 // 15 days

    fun sanitize(value: Long?): Long {
        if (value == null || value <= 0L) {
            return DEFAULT_LICENSE_EXPIRY_SECONDS
        }
        return minOf(value, DEFAULT_LICENSE_EXPIRY_SECONDS)
    }
}

File: ios/LicenseDurationUtils.swift
import Foundation

enum LicenseDurationUtils {
    static let DEFAULT_LICENSE_EXPIRY_SECONDS: Double = 15 * 24 * 60 * 60 // 15 days

    static func sanitize(_ value: Double) -> Double {
        guard value > 0 else { return DEFAULT_LICENSE_EXPIRY_SECONDS }
        return min(value, DEFAULT_LICENSE_EXPIRY_SECONDS)
    }
}
